titus-server-gateway/src/main/java/com/netflix/titus/gateway/service/v3/internal/DisruptionBudgetSanitizer.java

```diff
@@ -0,0 +1,139 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.titus.gateway.service.v3.internal;
+
+import java.util.Collections;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.netflix.spectator.api.Id;
+import com.netflix.spectator.api.Registry;
+import com.netflix.titus.api.jobmanager.model.job.JobDescriptor;
+import com.netflix.titus.api.jobmanager.model.job.JobFunctions;
+import com.netflix.titus.api.jobmanager.model.job.disruptionbudget.DisruptionBudget;
+import com.netflix.titus.api.jobmanager.model.job.disruptionbudget.DisruptionBudgetFunctions;
+import com.netflix.titus.api.jobmanager.model.job.disruptionbudget.SelfManagedDisruptionBudgetPolicy;
+import com.netflix.titus.api.jobmanager.model.job.disruptionbudget.UnlimitedDisruptionBudgetRate;
+import com.netflix.titus.api.jobmanager.model.job.ext.BatchJobExt;
+import com.netflix.titus.api.jobmanager.model.job.ext.ServiceJobExt;
+import com.netflix.titus.api.jobmanager.model.job.migration.MigrationPolicy;
+import com.netflix.titus.api.jobmanager.model.job.migration.SelfManagedMigrationPolicy;
+import com.netflix.titus.common.runtime.TitusRuntime;
+import com.netflix.titus.gateway.MetricConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Singleton
+public class DisruptionBudgetSanitizer {
+
+    private static final Logger logger = LoggerFactory.getLogger(DisruptionBudgetSanitizer.class);
+
+    private static final String METRICS_ROOT = MetricConstants.METRIC_JOB_MANAGEMENT + "disruptionBudget.";
+
+    private static final int MIN_BATCH_RELOCATION_TIME_MS = 60_000;
+
+    @VisibleForTesting
+    static final int DEFAULT_SERVICE_RELOCATION_TIME_MS = 60_000;
+
+    /**
+     * A strict alignment of a batch task runtime limit and a self migration deadline, may cause interruption of a batch task
+     * which is almost finished. To prevent that we set migration limit to a slightly higher value.
+     */
+    @VisibleForTesting
+    static final double BATCH_RUNTIME_LIMIT_FACTOR = 1.2;
+
+    private final DisruptionBudgetSanitizerConfiguration configuration;
+    private final Registry registry;
+
+    private final Id nonCompliantId;
+
+    @Inject
+    public DisruptionBudgetSanitizer(DisruptionBudgetSanitizerConfiguration configuration, TitusRuntime titusRuntime) {
+        this.configuration = configuration;
+        this.registry = titusRuntime.getRegistry();
+        this.nonCompliantId = registry.createId(METRICS_ROOT + "nonCompliant");
+    }
+
+    public JobDescriptor sanitize(JobDescriptor original) {
+        if (!DisruptionBudgetFunctions.isLegacyJobDescriptor(original)) {
+            return original;
+        }
+
+        if (!configuration.isEnabled()) {
```

Comment:
I don't think this toggle is needed anymore as everything going into the system should be rewritten going forward.